fdbrpc/RoleLineage.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * RoleLineage.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include "fdbrpc/Locality.h"
+
+struct RoleLineage : LineageProperties<RoleLineage> {
+    static StringRef name;
+    ProcessClass::ClusterRole role = ProcessClass::NoRole;
+
+    bool isSet(ProcessClass::ClusterRole RoleLineage::*member) {
```

Comment:
I don't understand this syntax. Do you have a reference I could look at for this (I'm not exactly sure what to google). What type does member have?

Reply:
This is the pointer to member syntax (we use this in `loadBalance`). The only thing I can find is MS documentation: https://docs.microsoft.com/en-us/cpp/cpp/pointer-to-member-operators-dot-star-and-star?view=msvc-160 (but if you google it you can also find some stack overflow questions that are useful)